app/models/tweet.rb

```diff
@@ -28,7 +28,12 @@ def liked_by?(user)
     likes.where(user:).any?
   end
 
-  has_many :visitors, dependent: :destroy
+  def add_visitor(current_user)
+    Visitor.find_or_create_by(tweet_id: id, user_id: current_user.id)
```

Comment:
@amnindersingh12 check how you can make use of association here

Reply:
Okay @swatijadhav